keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -584,6 +586,151 @@ would allow us to delete the pod and trying again with a new one, but is not don
 at the moment because admission checks cannot be retried if a check finds
 a transient problem.
 
+#### Partitionable devices
+
+In addition to devices, a `ResourceSlice` can also embed a list of
+`SharedCapacity` objects. Each `SharedCapacity` object represents some amount
+of shared "capacity" that can be consumed by one or more devices listed in the
+slice. When listing such devices, the sum of the capacity consumed across all
+devices *may* exceed the total amount available in the `SharedCapacity` object.
+This allows one, for example, to provide a way of logically partition a device
+into a set of overlapping sub-devices, each of which "could" be allocated by a
+scheduler (just not at the same time).
+
+As such, scheduler support will need to be added to track the set of
+`SharedCapacity` objects provided by each `ResourceSlice` as well as perform
+the following steps to decide if a given device is a candidate for allocation
+or not:
+
+1. Look at the set of `SharedCapacityConsumed` objects referenced by the device
+1. For each `SharedCapacityConsumed` object, look and see how much capacity is still available in the `SharedCapacity` object it is tracking with the same name
+1. If enough capacity is still available in the `SharedCapacity` object to satisfy the devices consumption, continue to consider it for allocation
+1. If not enough capacity is available, move on to the next device
+1. Upon deciding to allocate a device, subtract all of the capacity in its `SharedCapacityConsumed` objects from the corresponding `SharedCapacity` objects being tracked
+1. Upon freeing a device, add all of the capacity in its `SharedCapacityConsumed` objects back into the corresponding `SharedCapacity` objects being tracked
+
+**Note:** Only devices embedded in the _same_ `ResourceSlice` where a given
+`SharedCapacity` is declared have access to that `SharedCapacity`. This
+restriction simplifies the logic required to track these capacities in the
+scheduler, and shouldn't be too limiting in practice.
+
```

Comment:
Add something like this:

Also, note that while it is described as "subtracting" and "adding" capacity in the `SharedCapacity` object, these activities are not written back to the `ResourceSlice`. Just like all summaries of current allocations, they are maintained in-memory by the scheduler based on the totality of device allocations recorded in all claim statuses.

Reply:
Added